### PR TITLE
PLT-3030 Fix Blank Page for non-english languages

### PR DESCRIPTION
**Pull request by enahum**
Building first to see if it actually fixes the problem

### CODE REVIEW

**Comment by enahum** [comment]
This fixes the problem with the Blank Page when using a different locale than english confirmed with https://github.com/mattermost/platform/issues/3042

File: webapp/action_creators/global_actions.jsx [hunk 1]
```diff
@@ -392,7 +392,10 @@ export function newLocalizationSelected(locale) {
     } else {
         Client.getTranslations(
             I18n.getLanguageInfo(locale).url,
-            (data) => {
+            (data, res) => {
+                if (!data && res.text) {
+                    data = JSON.parse(res.text); //eslint-disable-line no-param-reassign
```
Comment (hmhealey) [inline webapp/action_creators/global_actions.jsx]: Maybe change this to the following? There's no point in having ESLint run if we're just disabling it on a line-by-line basis
```
let translations = data;
if (!data && res.text) {
    translations = JSON.parse(res.text);
}
```

**Comment by hmhealey** [comment]
What actually caused the blank page to show up? I'm kind of curious

File: webapp/action_creators/global_actions.jsx (hunk 1 above)
Comment (enahum) [inline webapp/action_creators/global_actions.jsx]: change submitted

**Comment by enahum** [comment]
The blank page was caused by the response of the [locale].json files, if the content-type returned was `application/json` everything worked as expected, but there where some cases when the return content-type was actually `plain/text` in that case the `res.body` was null and all the data returned was in `res.text`

**Comment by jwilander** [comment]
Should we not force all the translations to be returned with a Content-Type of `application/json` instead of trying to handle different cases?

**Comment by enahum** [comment]
@jwilander any proposal on how to do it server side?

**Comment by jwilander** [comment]
Hmmm, not sure. Just looked at the code and the driver seems to be adding all the correct headers and we're doing a straight GET to a static resource. I'm OK with doing this for now but we should probably figure this out, maybe it's an issue with superagent

**Comment by coreyhulen** [comment]
+1

**Comment by jwilander** [comment]
+1

**Pull request merged**